tv_apps/tv_shared/components/smart-dialog/script.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+window.SmartDialog = (function(win) {
+
+  // Extend from the HTMLElement prototype
+  var proto = Object.create(HTMLElement.prototype);
+
+  // Bezier curve function
+  var Bezier = function(bezierParam) {
```

Comment:
Also, may you add some comments describing this function? And it would be better if there are references for symbols here (a1, b1, c1..etc).